lib/utils/interval.go

```diff
@@ -0,0 +1,131 @@
+/*
+Copyright 2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import (
+	"errors"
+	"sync"
+	"time"
+)
+
+// Interval functions similarly to time.Ticker, with the added benefit of being
+// able to specify a custom duration for the first "tick", and an optional
+// per-tick jitter.  When attempting to stagger periodic operations it is recommended
+// to apply a large jitter to the first duration, and provide a small jitter for the
+// per-tick jitter.  This will ensure that operations started at similar times will
+// have varying initial interval states, while minimizing the amount of extra work
+// introduced by the per-tick jitter.
+type Interval struct {
```

Comment:
Code that needs backoff should use `utils.Retry` for that (we already use a combination of `time.Ticker` and `utils.Retry` in various parts of the codebase where we have periodics have a backoff component).  I'd prefer to keep `time.Ticker` and `interval.Interval` interchangeable as much as possible so that its easy to swap back to `time.Ticker` if the extra features of the interval are no longer needed.